### PR TITLE
added support for local (localStorage) CAT names

### PR DESCRIPTION
**Pull request by seeden**
[GUI: When naming a stray CAT wallet, don't add wallet until switch is turned on#1010](https://github.com/Chia-Network/chia-blockchain-gui/issues/1010)

### CODE REVIEW

File: packages/api-react/src/hooks/useLocalStorage.ts [hunk 1]
```diff
@@ -3,26 +3,32 @@ import EventEmitter from '../utils/EventEmitter';
...
-function getValueFromLocalStorage<T>(key: string, defaultValue: T) {
+function getValueFromLocalStorage<T>(
+  key: string,
+  defaultValue?: T
+): T | undefined {
   const item = window.localStorage.getItem(key);
 
   if (item === undefined || item === null) {
-    return defaultValue;
+    return undefined;
```
Comment (paninaro) [inline packages/api-react/src/hooks/useLocalStorage.ts:13]: This change appears to be causing an exception when loading WalletsSidebar with the useHiddenWallet hook.

I'm seeing this when selecting a key at launch (auto-login disabled):

image

Comment (paninaro) [reply packages/api-react/src/hooks/useLocalStorage.ts:13]: The issue seems to be that hiddenFingerprints is undefined:

image

Comment (paninaro) [reply packages/api-react/src/hooks/useLocalStorage.ts:13]: Seems to work if I add:

`
if (!hiddenWalletids) {
  return false;
}
`

Comment (paninaro) [reply packages/api-react/src/hooks/useLocalStorage.ts:13]: Hiding a wallet seems to throw an exception as well

Comment (seeden) [reply packages/api-react/src/hooks/useLocalStorage.ts]: I fixed typo in my code. Thanks for nice catch

**Review by paninaro — APPROVED** [review]
lgtm

**Pull request merged**